system/jlib/jlog.cpp

```diff
@@ -2391,22 +2391,25 @@ void setupContainerizedLogMsgHandler()
         if (logConfig->hasProp(logFormatAtt))
         {
             const char *logFormat = logConfig->queryProp(logFormatAtt);
-            bool newFormatDetected = false;
+            HandleLogMsgHandler *previousStderrHandler{nullptr};
             if (streq(logFormat, "xml"))
             {
+                previousStderrHandler = theStderrHandler;
                 theStderrHandler = new HandleLogMsgHandlerXML(stderr, MSGFIELD_STANDARD);
-                newFormatDetected = true;
             }
             else if (streq(logFormat, "json"))
             {
+                previousStderrHandler = theStderrHandler;
                 theStderrHandler = new HandleLogMsgHandlerJSON(stderr, MSGFIELD_STANDARD);
-                newFormatDetected = true;
             }
             else if (!streq(logFormat, "table"))
```

Comment:
NB: what if changing back to "table" format ? (should create a HandleLogMsgHandlerTable object).
But see other comments, don't change, assert that handler itself can't change.